tests/src/geometry_checker/testqgsgeometrychecks.cpp

```diff
@@ -329,6 +332,60 @@ void TestQgsGeometryChecks::testContainedCheck()
   cleanupTestContext( testContext );
 }
 
+void TestQgsGeometryChecks::testCurveCheck()
+{
+  QTemporaryDir dir;
+  QMap<QString, QString> layers;
+  layers.insert( QStringLiteral( "point_layer.shp" ), QString() );
+  layers.insert( QStringLiteral( "line_layer.shp" ), QString() );
+  layers.insert( QStringLiteral( "polygon_layer.shp" ), QString() );
+  layers.insert( QStringLiteral( "curved_line.gpkg" ), QString() );
+  layers.insert( QStringLiteral( "curved_polygon.gpkg" ), QString() );
+  const auto testContext = createTestContext( dir, layers );
+
+  // Test detection
+  QList<QgsGeometryCheckError *> checkErrors;
+  QStringList messages;
+
+  const QgsGeometryCurveCheck check( testContext.first, QVariantMap() );
+  QgsFeedback feedback;
+  check.collectErrors( testContext.second, checkErrors, messages, &feedback );
+  listErrors( checkErrors, messages );
+
+  QList<QgsGeometryCheckError *> errs1;
+  QList<QgsGeometryCheckError *> errs2;
+
+  QCOMPARE( checkErrors.size(), 4 );
+  QVERIFY( searchCheckErrors( checkErrors, layers["point_layer.shp"] ).isEmpty() );
+  QVERIFY( searchCheckErrors( checkErrors, layers["polygon_layer.shp"] ).isEmpty() );
+  QVERIFY( searchCheckErrors( checkErrors, layers["line_layer.shp"] ).isEmpty() );
+
+  QVERIFY( ( errs1 = searchCheckErrors( checkErrors, layers["curved_line.gpkg"], 2, QgsPointXY( 0.427921, 0.915679 ), QgsVertexId() ) ).size() == 1 );
+  QVERIFY( ( errs2 = searchCheckErrors( checkErrors, layers["curved_line.gpkg"], 3, QgsPointXY( 0.00970782, -0.225534 ), QgsVertexId() ) ).size() == 1 );
+  QVERIFY( searchCheckErrors( checkErrors, layers["curved_polygon.gpkg"], 2, QgsPointXY( -0.0659905, 0.312651 ), QgsVertexId() ).size() == 1 );
+  QVERIFY( searchCheckErrors( checkErrors, layers["curved_polygon.gpkg"], 3, QgsPointXY( -1.10633, -1.01106 ), QgsVertexId() ).size() == 1 );
+
+  // Test fixes
+  // Delete
+  QVERIFY( fixCheckError( testContext.second,  errs1[0],
+                          QgsGeometryCurveCheck::Delete, QgsGeometryCheckError::StatusFixed,
+  {{errs1[0]->layerId(), errs1[0]->featureId(), QgsGeometryCheck::ChangeFeature, QgsGeometryCheck::ChangeRemoved, QgsVertexId()}} ) );
+  QgsFeature f;
+  const bool valid = testContext.second[errs1[0]->layerId()]->getFeature( errs1[0]->featureId(), f );
+  QVERIFY( !valid );
+
+  // Convert to Straight segments
+  QVERIFY( fixCheckError( testContext.second,  errs2[0],
+                          QgsGeometryCurveCheck::StraightCurves, QgsGeometryCheckError::StatusFixed,
+  {{errs2[0]->layerId(), errs2[0]->featureId(), QgsGeometryCheck::ChangeFeature, QgsGeometryCheck::ChangeChanged, QgsVertexId()}} ) );
+  QgsFeature f2;
+  const bool valid2 = testContext.second[errs2[0]->layerId()]->getFeature( errs2[0]->featureId(), f2 );
+  QVERIFY( valid2 );
+  QVERIFY( ! f2.geometry().constGet()->hasCurvedSegments() );
+
+  cleanupTestContext( testContext );
+}
```

Comment:
Can you add another test to check the "do-nothing" case?

Reply:
Done in [0a9c00b](https://github.com/qgis/QGIS/pull/55590/commits/0a9c00b0cebd5e29ef0322f903a01d07b1d85a7a)